src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -576,7 +590,32 @@ private static void EnsureNullableAttributeExists(CSharpCompilation compilation,
                         break;
 
                     case SyntaxKind.ReadOnlyKeyword:
-                        diagnostics.Add(ErrorCode.ERR_ReadOnlyNotSuppAsParamModDidYouMeanIn, modifier.GetLocation());
+                        Binder.CheckFeatureAvailability(modifier, MessageID.IDS_FeatureRefReadonlyParameters, diagnostics);
```

Comment:
>Binder.CheckFeatureAvailability(modifier, MessageID.IDS_FeatureRefReadonlyParameters, diagnostics); [](http://example.com/codeflow?start=24&length=99)

Consider performing this check only if ```seenReadonly``` is false.

Reply:
Done, but it doesn't have much effect as `seenReadonly` is `false` when there are other errors (like missing `ref`).

Reply:
Instead, I can move the check where we set `seenReadonly` to `true` to reduce diagnostics even further, see ee2b4370b67.